tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -439,6 +448,8 @@ static PyObject * THPVariable_item(PyObject* self, PyObject* args)
     return wrap(dispatch_to_CDouble(self_));
   } else if (self_.is_complex()) {
     return wrap(dispatch_to_CComplexDouble(self_));
+  } else if (self_.scalar_type() == ScalarType::Bool) {
+    return wrap(dispatch_to_Bool(self_));
```

Comment:
what about __bool__?

Reply:
@gchanan, im not sure what about it? __bool__ is defined later in this file and can be called from python. I can't call THPVariable_bool directly from here.

Reply:
What I mean is THPVariable_bool has its own implementation that is probably specialized because we didn't have bool types before.  You can probably just have THPVariable_integral_scalar handle that case instead of a special case now.